ccswbs/hjckrrh
# Iss539 migrate format

Fixes #539 

In order to migrate all formats as filtered_html (ie. override the source format), the person migrating can go into their site-specific migration configuration file and:
- set the variable "default_format" to the format they wish to migrate to (eg. filtered_html)
- set the variable "override_format" to TRUE

![migrate-settings-filtered](https://cloud.githubusercontent.com/assets/1927902/24812823/a36d65fa-1b99-11e7-9e7b-4cf03b38b7a2.png)

Default behaviour remains to migrate the source format to the destination format (so if half the items on the source site are filtered_html and half are full_html, you will find the same on the migrated destination site). This allows us to migrate sites with their format intact, rather than setting a format which could potentially strip content if people saved.

---
# Demo

## Source site
![migrate-news-1-before](https://cloud.githubusercontent.com/assets/1927902/24812575/a726a9c8-1b98-11e7-9d25-9c782b2b0b61.png)

```

mysql> select body_format from field_data_body where bundle="article";
+---------------+
| body_format   |
+---------------+
| filtered_html |
| filtered_html |
| filtered_html |
| filtered_html |
| full_html     |
| plain_text    |
| full_html     |
| filtered_html |
| plain_text    |
| filtered_html |
| full_html     |
| filtered_html |
| full_html     |
| filtered_html |
| filtered_html |
| full_html     |
| plain_text    |
| full_html     |
| plain_text    |
| full_html     |
| filtered_html |
| plain_text    |
| full_html     |
| full_html     |
| filtered_html |
| full_html     |
| filtered_html |
| filtered_html |
| filtered_html |
| plain_text    |
| filtered_html |
| full_html     |
| filtered_html |
| filtered_html |
| plain_text    |
| plain_text    |
| custom_format |
| plain_text    |
| plain_text    |
| plain_text    |
| filtered_html |
| plain_text    |
| filtered_html |
| plain_text    |
| filtered_html |
| plain_text    |
| full_html     |
| full_html     |
| plain_text    |
| full_html     |
| filtered_html |
+---------------+
51 rows in set (0.00 sec)
```

## Destination Site
![migrate-news-2-after](https://cloud.githubusercontent.com/assets/1927902/24812659/ef7bf0ac-1b98-11e7-80f0-9df3a13c417a.png)

```
mysql> select field_news_body_format from field_data_field_news_body;
+------------------------+
| field_news_body_format |
+------------------------+
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
| filtered_html          |
+------------------------+
51 rows in set (0.00 sec)
```

---

# Testing Process
- open the site-specific migrate_settings configuration file
- set the variable "default_format" to filtered_html
- set the variable "override_format" to TRUE 
- migrate a site with a variety of source formats
- check in the database -- all your formats should now be filtered_html
